extension/chrome/settings/modules/my_key_update.ts

```diff
@@ -101,11 +103,14 @@ View.run(
     };
 
     private updatePrivateKeyHandler = async () => {
-      const updatedKey = await KeyUtil.parse(String(this.inputPrivateKey.val()));
       const updatedKeyEncrypted = await KeyUtil.parse(String(this.inputPrivateKey.val()));
+      const updatedKey = await KeyUtil.parse(KeyUtil.armor(updatedKeyEncrypted)); // create a "cloned" copy to decrypt later
       const updatedKeyPassphrase = String($('.input_passphrase').val());
+      KeyImportUi.allowReselect();
       if (typeof updatedKey === 'undefined') {
         await Ui.modal.warning(Lang.setup.keyFormattedWell(this.prvHeaders.begin, String(this.prvHeaders.end)), Ui.testCompatibilityLink);
+      } else if (updatedKeyEncrypted.identities.length === 0) {
+        await Ui.modal.error(Lang.setup.prvHasUseridIssue);
```

Comment:
This page is now using `KeyImportUi` for some visual handling, but it's not using `KeyImportUi.checkPrv` as it is not easy to adapt it here (we can try this in a separate issue)

Also, what should the message be if there is something wrong with UserIds (we currently don't support UserIds that don't contain email address in them)
I added this text
![image](https://github.com/FlowCrypt/flowcrypt-browser/assets/23335944/94fc7b0a-d60f-4a30-8dcd-c0b792cbda0b)

Reply:
@sosnovsky is there a vision on whether we should support keys with uids without emails?

Reply:
We currently support `identities` missing email address for public keys of recipients, but `compose` module assumes `emails[0]` matches `identities[0]` but that's not necessarily true. I suggest to refactor this part. Instead of "synchronized" arrays we can have an array of objects of the type returned by `Str.parseEmail`, it has `{ email, name, full }` -- everything that's required. `full` is the original `identity`

Reply:
> @sosnovsky is there a vision on whether we should support keys with uids without emails?

I think it's quite rare case and we didn't get any requests for such functionality, so we can leave it as is for now. Then we'll probably also need to update our mobile apps to support such keys.

> We currently support identities missing email address for public keys of recipients, but compose module assumes emails[0] matches identities[0] but that's not necessarily true. I suggest to refactor this part. 

That sounds good 👍